src/http_microhttpd.c

```diff
@@ -615,19 +627,26 @@ static int encode_and_redirect_to_splashpage(struct MHD_Connection *connection,
  */
 static int redirect_to_splashpage(struct MHD_Connection *connection, t_client *client, const char *host, const char *url)
 {
-	char *originurl = NULL;
+	const char *originurl = NULL;
 	char *query = NULL;
 	int ret = 0;
+	const char *token = NULL;
+	s_config *config = config_get_config();
 
 	get_query(connection, &query);
 	if (!query) {
 		/* no mem */
 		return send_error(connection, 503);
 	}
-
+	if (config->fas_secure_enabled != 1) {
+		safe_asprintf(&token, "%s", client->token);
+	} else {
+		safe_asprintf(&token, "%s&gatewayname=%s", client->ip, config->gw_name);
```

Comment:
token here is either the token or some http get argument. I do not think this is a good to sneak in extra http get variables.

Reply:
It is a bit of a hack - actually working on it as you commented - update coming soon.